plan/expression_rewriter.go

```diff
@@ -116,11 +149,16 @@ type expressionRewriter struct {
 	aggrMap  map[*ast.AggregateFuncExpr]int
 	b        *planBuilder
 	ctx      sessionctx.Context
-	// asScalar means the return value must be a scalar value.
+
+	// asScalar indicates the return value must be a scalar value.
+	// NOTE: This value can be changed during expression rewritten.
 	asScalar bool
 
 	// preprocess is called for every ast.Node in Leave.
 	preprocess func(ast.Node) ast.Node
+
+	inInsertOnDuplicateUpdate bool
+	insertPlan                *Insert
```

Comment:
Can we remove `inInsertOnDuplicateUpdate` and make a better name to `insertPlan`?~
And use  `[RenamedInsertPlan] != nil ` to confirm whether we use onDuplicate code path.

Reply:
Yes I can remove `inInsertOnDuplicateUpdate`, as for renaming the `insertPlan` variable, any suggestions?

Reply:
It's maybe not necessary to fix, use two field is ok, too (after 20+ days back to see this question- -)

Reply:
😂 I'll try to refine.